swap_meet/vendor.py

```diff
@@ -1,2 +1,167 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    """A class representing a vendor who wants to swap items with other vendors
+
+    Attributes:
+        inventory (list): a list of items to swap with other vendors.
+    """
+
+    def __init__(self, inventory = None):
+        """Initializes the vendor with given inventory or set to None as default.
+
+        Args:
+            inventory(list): a list of items            
+        """
+        self.inventory = [] if inventory is None else inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        """Swaps items between the current vendor and another vendor.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap items with.
+            my_item (Item): The item the current vendor wants to trade.
+            their_item (Item): The item the other vendor wants to trade.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+
+        """
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.remove(my_item)
+        other_vendor.remove(their_item)
+
+        other_vendor.add(my_item)
+        self.add(their_item)
+
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        """Swaps the first item of the current vendor's inventory with the first item of the other vendor's inventory.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap with.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+        """
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_item = self.inventory[0]
+        their_item = other_vendor.inventory[0]
+
+        return self.swap_items(other_vendor, my_item, their_item)
+
+    def get_by_category(self, category):
+        """Returns a list of items in the inventory that belong to a specific category.
+
+        Parameters:
+            category (str): The category to filter items by.
+
+        Returns:
+            list: A list of items that match the given category.
+        """
+        # item_list = []
+        # for item in self.inventory:
+        #     if item.get_category() == category:
+        #         item_list.append(item)
+        # use list comprehension to simplify code
+        return [item for item in self.inventory if item.get_category() == category]
```

Comment:
Great list comprehension here! I would go ahead and just make sure to remove your commented out and unused code!